Should hash_utils.h #include <cstddef>  (gcc11 compile failure)

hash_utils.h uses `size_t` but this is only defined by a `#include <cstddef>` in `hash_utils.cc`.
https://github.com/google/draco/blob/7e25df9c21bffb9aad89ecbea7677dfeadc791b0/src/draco/core/hash_utils.h#L26-L31

https://github.com/google/draco/blob/571d547d36a6546854948cfecdd7885a8bfb9c02/src/draco/core/hash_utils.cc#L15-L17

I have an R package that wraps draco (https://github.com/jefferis/dracor) and this fails to build on gcc 11 with an error like this:
```
[snip]
In file included from hash_utils.cc:15:
draco/core/hash_utils.h:27:1: error: 'size_t' does not name a type
   27 | size_t HashCombine(T1 a, T2 b) {
      | ^~~~~~
draco/core/hash_utils.h:21:1: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
   20 | #include <functional>
  +++ |+#include <cstddef>
   21 | 
draco/core/hash_utils.h:34:1: error: 'size_t' does not name a type
   34 | size_t HashCombine(T a, size_t hash) {
      | ^~~~~~
draco/core/hash_utils.h:34:1: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
draco/core/hash_utils.h:44:43: error: 'size_t' has not been declared
   44 | uint64_t FingerprintString(const char *s, size_t len);
      |                                           ^~~~~~
draco/core/hash_utils.h:49:3: error: 'size_t' does not name a type
   49 |   size_t operator()(const T &a) const {
      |   ^~~~~~
draco/core/hash_utils.h:49:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
draco/core/hash_utils.h:58:3: error: 'size_t' does not name a type
   58 |   size_t ValueHash(const V &val) const {
      |   ^~~~~~
draco/core/hash_utils.h:58:3: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
[snap]
```
I think it would make sense to move the `#include <cstddef>` into `hash_utils.h`. Thank you!

--- a/src/draco/core/hash_utils.h
+++ b/src/draco/core/hash_utils.h
@@ -17,9 +17,8 @@
 
 #include <stdint.h>
 
+#include <cstddef>
 #include <functional>
-
-// TODO(fgalligan): Move this to core.
 
 namespace draco {
 

